pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -217,7 +220,17 @@ func readInstanceID() (string, error) {
 }
 
 func newOpenStack(cfg Config) (*OpenStack, error) {
-	provider, err := openstack.AuthenticatedClient(cfg.toAuthOptions())
+	provider, err := openstack.NewClient(cfg.Global.AuthUrl)
+	if cfg.Global.TrustId != "" {
```

Comment:
Indeed. For the new function vs actual what do you prefer reviewers ? I don't care on my side, I saw a previous point from @liggitt in the other review regarding the fact that the function is exported, but we can just not export it.